src/jarabe/frame/activitiestray.py

```diff
@@ -90,6 +109,44 @@ def __init__(self, home_activity, group):
         elif home_activity.props.launch_status == shell.Activity.LAUNCH_FAILED:
             self._on_failed_launch()
 
+    def __drag_drop_cb(self, widget, context, x, y, time):
+        context_targets = context.list_targets()
+        if not self._context_map.has_context(context):
+            self._context_map.add_context(context, 0, len(context_targets))
+
+        for target in context_targets:
+            if str(target) not in ('TIMESTAMP', 'TARGETS', 'MULTIPLE'):
+                widget.drag_get_data(context, target, time)
+
+        cb_menu = ClipboardMenu(self._cb_object)
+        cb_menu._copy_to_journal()
+
+        logging.debug('Saved to Journal')
+        return True
+
+    def __journal_drag_data_received_cb(self, widget, context, x, y, selection,
+                                        targetType, time):
+        data = None
+        if selection.get_pixbuf() is not None:
+            data = selection.get_pixbuf()
+        if len(selection.get_uris()) != 0:
+            data = selection.get_uris()[0].encode()
+            GLib.strfreev(data)
```

Comment:
What does L133-134 do?

Reply:
It basically gets the first item from a list of URIs and encodes that data for `os.write`, [`GLib.strfreev`](https://lazka.github.io/pgi-docs/index.html#Gtk-3.0/classes/SelectionData.html#Gtk.SelectionData.get_uris) is called if the result is non-None and in this case it's expected that it's non-None.